jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java

```diff
@@ -186,4 +191,40 @@ private void publishList(XmlUtils xmlUtils, List<OdeData> dataList) throws JsonP
 		}
 	}
 
+	public String checkHeader(OdeMsgPayload payload) {
+		JSONObject payloadJson;
+		String header = null;
+		try {
+			payloadJson = JsonUtils.toJSONObject(payload.getData().toJson());
+			String hexPacket = payloadJson.getString("bytes");
+
+			for (String key : msgStartFlags.keySet()) {
+				String startFlag = msgStartFlags.get(key);
+				int startIndex = hexPacket.toLowerCase().indexOf(startFlag);
+				logger.debug("Start index for " + key + "(" + startFlag + ")" + " is: " + startIndex);
+				if (startIndex <= 20 && startIndex != 0 && startIndex != -1) {
+					logger.debug("Message has supported Ieee1609Dot2Data header, adding encoding rule to Asn1DecoderInput XML");
+					header = "Ieee1609Dot2Data";
```

Comment:
If we've found the header here, should we break out of the loop to return? Is there any reason to continue at that point?

Reply:
Nope, good idea!